app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/CalendarDetailsFragment.kt

```diff
@@ -46,24 +48,46 @@ class CalendarDetailsFragment : RoundedBottomSheetDialogFragment() {
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
-        val calendarItem = dao.getCalendarItemById(calendarId)
+        val calendarItem = dao.getCalendarItemsById(calendarId)
         updateView(calendarItem)
     }
 
-    private fun updateView(calendarItem: CalendarItem) {
+    private fun updateView(calendarItemList: List<CalendarItem>) {
+        val calendarItem = calendarItemList[0]
+
         if (calendarItem.status == "CANCEL") {
             cancelButtonsContainer.visibility = View.VISIBLE
             descriptionTextView.setTextColor(Color.RED)
         }
 
         titleTextView.text = calendarItem.title
         dateTextView.text = calendarItem.getEventDateString()
-        locationTextView.text = calendarItem.location
 
-        if (calendarItem.location.isEmpty()) {
-            locationTextView.visibility = View.GONE
+        val locationList = toLocationsList(calendarItemList)
+        if (areLocationsEmpty(locationList)) {
+            locationIcon.visibility = View.GONE
         } else {
-            locationTextView.visibility = View.VISIBLE
+            locationIcon.visibility = View.VISIBLE
+            for ((index, item) in calendarItemList.withIndex()) {
+                if (item.location.isBlank()) {
+                    continue
+                }
+                val locationText = TextView(context)
+                if (item.isCancelled()) {
+                    locationText.setTextColor(resources.getColor(R.color.event_canceled))
+                    locationText.text = "${item.location} (${R.string.event_canceled})"
+                } else {
+                    locationText.setTextColor(resources.getColor(R.color.location_color))
+                    locationText.text = item.location
+                }
+                // Add padding between locations.
+                if (index + 1 < calendarItemList.size) {
+                    locationText.setPadding(0,0,0, dimen(R.dimen.material_small_padding))
+                }
+                locationText.textSize = 18f
```

Comment:
Could we extract that TextView into an inflatable XML layout to have a separation of logic and design?